settings/src/components/tests/password/password.test.js

```diff
@@ -0,0 +1,177 @@
+			// State: the user has updated their password to something weak
+			// although the strength is not tested here.
+			mockContext.userRecord.editedRecord.password = 'weak';
+			mockContext.userRecord.hasEdits = true;
+
+			useContext.mockReturnValue( mockContext );
+
+			// We'll mock the password estimator to return a score of 1.
+			mockPasswordEstimator( 1 );
+
+			const { queryAllByText } = render( <Password />, {
+				wrapper: ( { children } ) => (
+					<GlobalContext.Provider value={ mockContext }>
+						{ children }
+					</GlobalContext.Provider>
+				),
+			} );
+
+			// We may have this message multiple times, because of 'speak'.
+			expect(
+				queryAllByText( weakPasswordRegex ).length
+			).toBeGreaterThanOrEqual( 1 );
+		} );
+
+		it( 'should display the strong password notice', () => {
+			// State: the user has updated their password to something strong
+			// although the strength is not tested here.
+			mockContext.userRecord.editedRecord.password = '@#4asdf34asdfasdf';
+			mockContext.userRecord.hasEdits = true;
+
+			useContext.mockReturnValue( mockContext );
+
+			// We'll mock the password estimator to return a score of 10.
+			mockPasswordEstimator( 10 );
+
+			const { queryAllByText } = render( <Password />, {
+				wrapper: ( { children } ) => (
+					<GlobalContext.Provider value={ mockContext }>
+						{ children }
+					</GlobalContext.Provider>
+				),
+			} );
+
+			// We may have this message multiple times, because of 'speak'.
+			expect(
+				queryAllByText( strongPasswordRegex ).length
+			).toBeGreaterThanOrEqual( 1 );
+		} );
+	} );
+
+	it( 'should update the user record password', () => {
+		useContext.mockReturnValue( mockContext );
+
+		const { getByLabelText } = render( <Password />, {
+			wrapper: ( { children } ) => (
+				<GlobalContext.Provider value={ mockContext }>
+					{ children }
+				</GlobalContext.Provider>
+			),
+		} );
+
+		const input = getByLabelText( 'New Password' );
+		const newPassword = 'this is a password';
+		fireEvent.change( input, { target: { value: newPassword } } );
+		expect( mockContext.userRecord.edit ).toHaveBeenCalledWith( {
+			password: newPassword,
+		} );
+	} );
+
+	it( 'should submit form on button press', () => {
+		// State: the user has updated their password to something strong
+		// although the strength is not tested here.
+		mockContext.userRecord.editedRecord.password = '@#4asdf34asdfasdf';
+		mockContext.userRecord.hasEdits = true;
+
+		useContext.mockReturnValue( mockContext );
+
+		// We'll mock the password estimator to return a score of 10.
+		mockPasswordEstimator( 10 );
+
+		const { getAllByRole } = render( <Password />, {
+			wrapper: ( { children } ) => (
+				<GlobalContext.Provider value={ mockContext }>
+					{ children }
+				</GlobalContext.Provider>
+			),
+		} );
+
+		const buttons = getAllByRole( 'button' );
+		const saveButton = buttons.filter(
+			( button ) => button.type === 'submit'
+		)[ 0 ];
+		fireEvent.click( saveButton );
```

Comment:
Is there a way to also test the `enter` key?

Reply:
Usually there is, but as I mention in the description, I couldn't get it to work :). Which is ironic because it's what it want to test. 

I would be able to do it using [enzyme](https://enzymejs.github.io/enzyme/) but Gutenberg recently removed it from their project and while I didn't look into why they chose, I would assume we should follow suit. It's better to not mount/shallow render components anyways. 

With that being said, here are things that didn't work.

- Direct: Spy on the prototype, ie `spyOn( Password.prototype, 'formSubmitHandler' )`
  - Unfortunately a functional component doesn't have a prototype
- Indirect: Tried to track calls of `userRecord.save` since it's in the `formSubmitHandler` function.
  - While `formSubmitHandler` would get called this mock would always be empty
    - I assumed it was due to the async call not returning a promise, but even after attempting that it still would show no calls. I admittedly gave up.

Ideally, the form is a component and we pass in the onSubmit handler.  We could just pass a mock `<Form onFormSubmit={ mockCallback } />` and test the call, but I didn't want to restructure this component in this pass.

Reply:
What is bizarre though is that clicking the save button does register calls on the save function. 🤔 Maybe I'll take a look with fresh eyes.

Reply:
Ah, nice find!

I've experienced that w/ Copilot too. Just like w/ ChatGPT, I have to spend time verifying that the solution it gives is actually correct. It still saves some time in some cases though.